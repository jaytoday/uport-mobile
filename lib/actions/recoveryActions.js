// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  SHOW_RECOVERY_SEED,
  SAVE_RECOVERY_SEED,
  HIDE_RECOVERY_SEED,
  SELECT_RECOVERY_SEED_WORD,
  START_SEED_RECOVERY,
  RECOVERY_SUCCESS
} from 'uPortMobile/lib/constants/RecoveryActionTypes'
// import { SOCIAL_RECOVERY_POP_ROUTE, SOCIAL_RECOVERY_PUSH_ROUTE, SOCIAL_RECOVERY_RESET_ROUTE } from '../constants/SocialRecoveryActionTypes'

export function showRecoverySeed () {
  return {
    type: SHOW_RECOVERY_SEED
  }
}

export function saveRecoverySeed (seed) {
  return {
    type: SAVE_RECOVERY_SEED,
    seed
  }
}

export function hideRecoverySeed () {
  return {
    type: HIDE_RECOVERY_SEED
  }
}

export function selectRecoveryWord (index) {
  return {
    type: SELECT_RECOVERY_SEED_WORD,
    index
  }
}

export function startSeedRecovery (phrase, componentId) {
  return {
    type: START_SEED_RECOVERY,
    phrase,
    componentId
  }
}

export function recoverySuccess () {
  return {
    type: RECOVERY_SUCCESS
  }
}
