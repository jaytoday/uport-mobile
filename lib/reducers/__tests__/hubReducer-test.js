// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

import reducer from 'uPortMobile/lib/reducers/hubReducer.js'
import { storeHubHead, queueEvent, takeQueuedEvent, clearQueue } from 'uPortMobile/lib/actions/hubActions.js'
import { startSeedRecovery } from 'uPortMobile/lib/actions/recoveryActions.js'

const empty = { events: [] }
const doNotQueue = { events: [], doNotQueue: true }
const withData = { head: 'bcdefg', events: [{type: 'HELLO'}] }
const withQueue = { head: 'bcdefg', events: [{ type: 'HELLO1' }, { type: 'HELLO2' }, { type: 'HELLO3' }] }
const event = {type: 'BOOM', name: 'Quirp'}

it('sets head', () => {
  expect(reducer(undefined, storeHubHead('abcde'))).toMatchSnapshot()
  expect(reducer(empty, storeHubHead('abcde'))).toMatchSnapshot()
  expect(reducer(withData, storeHubHead('abcde'))).toMatchSnapshot()
  expect(reducer(doNotQueue, storeHubHead('abcde'))).toMatchSnapshot()
})

it('queues event', () => {
  expect(reducer(undefined, queueEvent(event))).toMatchSnapshot()
  expect(reducer(empty, queueEvent(event))).toMatchSnapshot()
  expect(reducer(withData, queueEvent(event))).toMatchSnapshot()
  expect(reducer(withQueue, queueEvent(event))).toMatchSnapshot()
})

it('takes queued event', () => {
  expect(reducer(undefined, takeQueuedEvent())).toMatchSnapshot()
  expect(reducer(empty, takeQueuedEvent())).toMatchSnapshot()
  expect(reducer(withData, takeQueuedEvent())).toMatchSnapshot()
  expect(reducer(withQueue, takeQueuedEvent())).toMatchSnapshot()
})

it('clears queue', () => {
  expect(reducer(undefined, clearQueue(event))).toMatchSnapshot()
  expect(reducer(empty, clearQueue(event))).toMatchSnapshot()
  expect(reducer(withData, clearQueue(event))).toMatchSnapshot()
  expect(reducer(withQueue, clearQueue(event))).toMatchSnapshot()
})

it('marks as doNotQueue', () => {
  expect(reducer(undefined, startSeedRecovery())).toMatchSnapshot()
  expect(reducer(empty, startSeedRecovery())).toMatchSnapshot()
  expect(reducer(withQueue, startSeedRecovery())).toMatchSnapshot()
})

it('ignores unsupported action', () => {
  expect(reducer(undefined, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(reducer(empty, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(reducer(withData, {type: 'UNSUPPORTED'})).toEqual(withData)
})
