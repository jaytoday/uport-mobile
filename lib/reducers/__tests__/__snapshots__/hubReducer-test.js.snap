// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears queue 1`] = `
Object {
  "events": Array [],
}
`;

exports[`clears queue 2`] = `
Object {
  "events": Array [],
}
`;

exports[`clears queue 3`] = `
Object {
  "events": Array [],
  "head": "bcdefg",
}
`;

exports[`clears queue 4`] = `
Object {
  "events": Array [],
  "head": "bcdefg",
}
`;

exports[`marks as doNotQueue 1`] = `
Object {
  "doNotQueue": true,
  "events": Array [],
}
`;

exports[`marks as doNotQueue 2`] = `
Object {
  "doNotQueue": true,
  "events": Array [],
}
`;

exports[`marks as doNotQueue 3`] = `
Object {
  "doNotQueue": true,
  "events": Array [
    Object {
      "type": "HELLO1",
    },
    Object {
      "type": "HELLO2",
    },
    Object {
      "type": "HELLO3",
    },
  ],
  "head": "bcdefg",
}
`;

exports[`queues event 1`] = `
Object {
  "events": Array [
    Object {
      "name": "Quirp",
      "type": "BOOM",
    },
  ],
}
`;

exports[`queues event 2`] = `
Object {
  "events": Array [
    Object {
      "name": "Quirp",
      "type": "BOOM",
    },
  ],
}
`;

exports[`queues event 3`] = `
Object {
  "events": Array [
    Object {
      "type": "HELLO",
    },
    Object {
      "name": "Quirp",
      "type": "BOOM",
    },
  ],
  "head": "bcdefg",
}
`;

exports[`queues event 4`] = `
Object {
  "events": Array [
    Object {
      "type": "HELLO1",
    },
    Object {
      "type": "HELLO2",
    },
    Object {
      "type": "HELLO3",
    },
    Object {
      "name": "Quirp",
      "type": "BOOM",
    },
  ],
  "head": "bcdefg",
}
`;

exports[`sets head 1`] = `
Object {
  "events": Array [],
  "head": "abcde",
}
`;

exports[`sets head 2`] = `
Object {
  "events": Array [],
  "head": "abcde",
}
`;

exports[`sets head 3`] = `
Object {
  "events": Array [
    Object {
      "type": "HELLO",
    },
  ],
  "head": "abcde",
}
`;

exports[`sets head 4`] = `
Object {
  "events": Array [],
  "head": "abcde",
}
`;

exports[`takes queued event 1`] = `
Object {
  "events": Array [],
}
`;

exports[`takes queued event 2`] = `
Object {
  "events": Array [],
}
`;

exports[`takes queued event 3`] = `
Object {
  "events": Array [],
  "head": "bcdefg",
}
`;

exports[`takes queued event 4`] = `
Object {
  "events": Array [
    Object {
      "type": "HELLO2",
    },
    Object {
      "type": "HELLO3",
    },
  ],
  "head": "bcdefg",
}
`;
